application/classes/Ushahidi/Repository/Post.php

```diff
@@ -138,20 +138,42 @@ protected function getPostValues($id)
 		// Get all the values for the post. These are the EAV values.
 		$values = $this->post_value_factory
 			->proxy($this->include_value_types)
-			->getAllForPost($id, $this->include_attributes);
-
+				->getAllForPost($id, $this->include_attributes);
+		
+		$messageLocation = 'message_location';
+        $found_location = false;
 		$output = [];
+
 		foreach ($values as $value) {
+			/*  check to see if there are other locations set than message_location */
+			if($this->checkForLocation($value, $messageLocation)) 
+			{
```

Comment:
I think this looks reasonable but it would be good to get a second opinion from @rjmackay and @jasonmule